ui/qt-lib/src/Ariadne/UI/Qt/Widgets/Dialogs/Settings.hs

```diff
@@ -0,0 +1,270 @@
+import Graphics.UI.Qtah.Signal (connect_)
+import Graphics.UI.Qtah.Widgets.QSizePolicy (QSizePolicyPolicy(..))
+
+import qualified Graphics.UI.Qtah.Core.QObject as QObject
+import qualified Graphics.UI.Qtah.Widgets.QAbstractButton as QAbstractButton
+import qualified Graphics.UI.Qtah.Widgets.QBoxLayout as QBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QComboBox as QComboBox
+import qualified Graphics.UI.Qtah.Widgets.QDialog as QDialog
+import qualified Graphics.UI.Qtah.Widgets.QHBoxLayout as QHBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QLabel as QLabel
+import qualified Graphics.UI.Qtah.Widgets.QLayout as QLayout
+import qualified Graphics.UI.Qtah.Widgets.QPushButton as QPushButton
+import qualified Graphics.UI.Qtah.Widgets.QVBoxLayout as QVBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QWidget as QWidget
+
+import Ariadne.UI.Qt.UI
+import Ariadne.UI.Qt.Widgets.Dialogs.Util
+import Ariadne.Util
+import Ariadne.Version (currentAriadneVersion)
+
+data Settings =
+  Settings
+    { settings :: QDialog.QDialog
+    , generalButton :: QPushButton.QPushButton
+    , aboutButton :: QPushButton.QPushButton
+    , supportButton :: QPushButton.QPushButton
+    , generalWidget :: QWidget.QWidget
+    , generalSettings :: GeneralSettings
+    , aboutWidget :: QWidget.QWidget
+    , aboutSettings :: AboutSettings
+    , supportWidget :: QWidget.QWidget
+    , supportSettings :: SupportSettings
+    }
+
+data SettingsMode = GeneralMode | AboutMode | SupportMode deriving (Show, Eq, Enum, Bounded)
+
+data GeneralSettings =
+  GeneralSettings
+    { languageSelector :: QComboBox.QComboBox
+    , countervalueSelector :: QComboBox.QComboBox
+    , rateProviderSelector :: QComboBox.QComboBox
+    , themeSelector :: QComboBox.QComboBox
+    }
+
+data AboutSettings =
+  AboutSettings
+    { releaseNotesButton :: QPushButton.QPushButton
+    , licenseButton :: QPushButton.QPushButton
+    }
+
+data SupportSettings =
+  SupportSettings
+    { faqButton :: QPushButton.QPushButton
+    , clearCacheButton :: QPushButton.QPushButton
+    , resetButton :: QPushButton.QPushButton
+    }
+
+makeLensesWith postfixLFields ''Settings
+
+initSettings :: IO (QDialog.QDialog, Settings)
+initSettings = do
+  settings <- QDialog.new
+  QWidget.resizeRaw settings 776 363
+
+  QObject.setObjectName settings ("settingsDialog" :: String)
+  QWidget.setWindowTitle settings ("Settings" :: String)
+  QWidget.adjustSize settings
+
+  settingsLayout <- createLayout settings
+  QLayout.setContentsMarginsRaw settingsLayout 0 24 0 0
+  QLayout.setSpacing settingsLayout 0
+
+  label <- QLabel.newWithText ("SETTINGS" :: String)
+
+  addHeader settingsLayout label
+  void $ QLayout.setWidgetAlignment settingsLayout label $ alignHCenter .|. alignVCenter
+  void $ setProperty label ("styleRole" :: Text) ("dialogHeader" :: Text)
+  buttonsLayout <- QHBoxLayout.new
+  QBoxLayout.setSpacing buttonsLayout 6
+  QLayout.setContentsMarginsRaw buttonsLayout 202 0 202 0
```

Comment:
Is this to make buttons centered? If so, please call `QBoxLayout.addStretch` before first and after last button.

Reply:
this is to make indentation, which is mentioned in zeplin

Reply:
Try resizing the dialog and see what happens to the buttons. AFAIU, doing what I suggested would fix that.